src/entities/DFGVariable.jl

```diff
@@ -10,13 +14,15 @@ mutable struct VariableNodeData
   eliminated::Bool
   BayesNetVertID::Symbol #  Union{Nothing, }
   separator::Array{Symbol,1}
-  softtype
+  softtype::InferenceVariable #TODO v0.5.0 T
   initialized::Bool
   inferdim::Float64
   ismargin::Bool
   dontmargin::Bool
+  # Tonio surprise TODO
+  # frontalonly::Bool
   # A valid, packable default constructor is needed.
-  VariableNodeData() = new(zeros(1,1), zeros(1,1), Symbol[], Int[], 0, false, :NOTHING, Symbol[], "", false, false, false, false)
+  VariableNodeData() = new(zeros(1,1), zeros(1,1), Symbol[], Int[], 0, false, :NOTHING, Symbol[], SingletonInferenceVariable(), false, false, false, false)
```

Comment:
The use of `Singleton` here is not that great, but can fix after v0.4.1.  `Singleton` relates to Factors not Variables

Reply:
For future reference, for single dimension variable or default, I use `ContinuousScalar`.

Reply:
I just named it Singleton from: Immutable composite types with no fields are singletons; 
I’m open to any suggestions. Default, empty.. etc. 
It should ideally only be temporary In use until the softtype becomes type parameter. 
Do you propose we move or duplicate ContinuousScalar to DFG?